ui/src/app/dashboard/grouped-summary/grouped-summary.component.ts

```diff
@@ -1,49 +1,130 @@
+      this.aggregationEntries.push(row);
+    }
+    // make a copy of original data
+    this.unsortedAggregationEntries = this.aggregationEntries.slice();
+
+    if (this.unsortedAggregationEntries.length > this.minDisplayNumRow) {
+      this.displayCollapseArrow = true;
+      this.paddingBottom = '0.2rem';
     }
-    return countMap;
   }
 
-  getLabelUrlParams(entry: AggregationEntry) {
+  private convertToStatusCountsMap(entry: AggregationEntry) {
+    let statusCountsMap = new Map();
+    for (let countEntry of entry.statusCounts.counts) {
+      statusCountsMap.set(countEntry.status, countEntry.count);
+    }
+    return statusCountsMap;
+  }
+
+  getLabelUrlParams(entry) {
     let map = this.getCommonUrlParamsMap(entry);
     return {q: URLSearchParamsUtils.encodeURLSearchParamsFromMap(map)};
   }
 
-  getStatusUrlParams(entry: AggregationEntry, status: JobStatus) {
+  getStatusUrlParams(entry, status: JobStatus) {
     let map = this.getCommonUrlParamsMap(entry);
     map.set('statuses', [status.toString()]);
     return {q: URLSearchParamsUtils.encodeURLSearchParamsFromMap(map)};
   }
 
-  getCommonUrlParamsMap(entry: AggregationEntry) {
+  getCommonUrlParamsMap(entry) {
     let map = new Map<string, string[]>();
-
     const projectId = URLSearchParamsUtils.unpackURLSearchParams(this.activatedRoute.snapshot.queryParams['q'])
       .extensions['projectId'];
 
     map.set('projectId', [projectId]);
-    map.set(this.aggregation.key, [entry.label]);
+    map.set(this.aggregation.key, [entry.get('label')]);
 
     return map;
   }
+
+  getStatusIcon(status: JobStatus): string {
+    return JobStatusIcon[status];
+  }
+
+  sortData(sort: Sort) {
+    const data = this.unsortedAggregationEntries.slice();
+    if (!sort.active || sort.direction === '') {
+      this.aggregationEntries = data;
+      return;
+    }
+
+    this.aggregationEntries = data.sort((a, b) => {
+      const isAsc = sort.direction === 'asc';
+      switch (sort.active) {
+        case 'label': return compare(a.get('label'), b.get('label'), isAsc);
+        case 'Running': return compare(a.get('Running'), b.get('Running'), isAsc);
+        case 'Aborted': return compare(a.get('Aborted'), b.get('Aborted'), isAsc);
+        case 'Succeeded': return compare(a.get('Succeeded'), b.get('Succeeded'), isAsc);
+        case 'Failed': return compare(a.get('Failed'), b.get('Failed'), isAsc);
+        case 'Submitted': return compare(a.get('Submitted'), b.get('Submitted'), isAsc);
+        case 'Aborting': return compare(a.get('Aborting'), b.get('Aborting'), isAsc);
+        default: return 0;
+      }
+    });
+  }
+
+  onArrowClick() {
+    this.expand = !this.expand;
+    if (this.expand) {
+      this.iconTransform = 'rotate(0deg)';
+      this.displayNumRow = this.unsortedAggregationEntries.length;
+    } else {
+      this.displayNumRow = DEFAULT_NUM_ROW;
+      this.iconTransform = 'rotate(180deg)';
+    }
+  }
+
+}
+
+function compare(a, b, isAsc) {
```

Comment:
nit: might be more appropriate to define this as a private method

Reply:
Accepted.